src/tezos_fmt_plug.c

```diff
@@ -0,0 +1,343 @@
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "jumbo.h"
+#include "ed25519.h"
+#include "blake2.h"
+#define PBKDF2_HMAC_SHA512_VARYING_SALT 1
+#include "pbkdf2_hmac_sha512.h"
+#include "memdbg.h"
+
+#define FORMAT_NAME             "Tezos Key"
+#define FORMAT_LABEL            "tezos"
+#define FORMAT_TAG              "$tezos$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME		"PBKDF2-SHA512 " SHA512_ALGORITHM_NAME
+#else
+#if ARCH_BITS >= 64
+#define ALGORITHM_NAME          "PBKDF2-SHA512 64/" ARCH_BITS_STR " " SHA2_LIB
+#else
+#define ALGORITHM_NAME          "PBKDF2-SHA512 32/" ARCH_BITS_STR " " SHA2_LIB
+#endif
+#endif
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        -1
+#define BINARY_SIZE             0
+#define BINARY_ALIGN            1
+#define SALT_SIZE               sizeof(struct custom_salt)
+#define SALT_ALIGN              sizeof(uint32_t)
+#define PLAINTEXT_LENGTH        125
+#ifdef SIMD_COEF_64
+#define MIN_KEYS_PER_CRYPT      SSE_GROUP_SZ_SHA512
+#define MAX_KEYS_PER_CRYPT      (SSE_GROUP_SZ_SHA512 * 4)
+#else
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      4
+#endif
+
+static struct fmt_tests tezos_tests[] = {
+	// http://doc.tzalpha.net/introduction/zeronet.html, https://faucet.tzalpha.net/
+	{"$tezos$1*2048*put guide flat machine express cave hello connect stay local spike ski romance express brass*jbzbdybr.vpbdbxnn@tezos.example.org*tz1eTjPtwYjdcBMStwVdEcwY2YE3th1bXyMR*a19fce77caa0729c68072dc3eb274c7626a71880d926", "4FGU8MpuCo"},
+	{"$tezos$1*2048*shove average clap front casino lawn segment dinosaur early solve hole dinner copy journey alley*kqdbxkwa.xvlnjlhg@tezos.example.org*tz1ZRcC58RDjA17Jmp2zDds6Hnk8UAjU8sxh*a19f97385132d6051136ef34d6a62a0bf5af9fecbe26", "XRknDmWXTm"},
+	{NULL}
+};
+
+static char (*saved_key)[PLAINTEXT_LENGTH + 1];
+static int *cracked, cracked_count;
+
+static struct custom_salt {
+	uint32_t type;
+	uint32_t iterations;
+	uint32_t email_length;
+	uint32_t mnemonic_length;
+	uint32_t raw_address_length;
+	char mnemonic[512];
+	char email[256];
+	char address[64];
+	char raw_address[64];
+} *cur_salt;
+
+static void init(struct fmt_main *self)
+{
+	omp_autotune(self, OMP_SCALE);
+
+	saved_key = mem_calloc(sizeof(*saved_key), self->params.max_keys_per_crypt);
+	cracked = mem_calloc(sizeof(*cracked), self->params.max_keys_per_crypt);
+	cracked_count = self->params.max_keys_per_crypt;
+}
+
+static void done(void)
+{
+	MEM_FREE(cracked);
+	MEM_FREE(saved_key);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+	char *ctcopy, *keeptr, *p;
+	int value, extra;
+
+	if (strncmp(ciphertext, FORMAT_TAG, TAG_LENGTH) != 0)
+		return 0;
+
+	ctcopy = strdup(ciphertext);
+	keeptr = ctcopy;
+
+	ctcopy += TAG_LENGTH;
+	if ((p = strtokm(ctcopy, "*")) == NULL) // type
+		goto err;
+	if (!isdec(p))
+		goto err;
+	value = atoi(p);
+	if (value != 1)
+		goto err;
```

Comment:
`strcmp(p, "1")` would be easier and stricter (would eliminate the need for `isdec()`).